### DISCUSSION TITLE
多次渲染造成内存泄露

### DISCUSSION THREAD

**Discussion by zmcode** [antvis/G6#4465] [Q&A]
这个视图是通过弹窗打开的, 每次关闭弹窗我都会销魂canvas, 删除这个dom元素, 可是内存不能得到释放, 每次打开, 又会不断增加内存
该如何优化
![image](https://user-images.githubusercontent.com/48686959/234032584-3e59c808-882d-43d4-8eb4-b604a39038b9.png)
![image](https://user-images.githubusercontent.com/48686959/234032798-2c3f0822-d3eb-4338-b4a1-3cc3e2db5466.png)
看着像是节点数据
我是通过一个函数来渲染数据的, 按道理来说不是赢回释放内存的吗, 有懂的能说说,吗

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/234034801-f009371d-5b82-48a6-9467-6361560b6d39.png)

**Comment by zmcode** [comment]
实现大概就是现在注册了节点, 然后一个函数里面定义了一个node和edge变量, 里面循环数据, 生成好node和edge数据, 然后render, 关闭弹窗, 删除节点数据, 打开弹窗又继续执行上面的操作, 尝试了下, 打开10次左右, 内存占用100MB了

**Comment by zmcode** [comment]
![1682348820197](https://user-images.githubusercontent.com/48686959/234038020-3dcdde51-0eb8-4bb0-b79f-5a1b9b960595.png)
似乎是这些对象没有得到释放

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/234038828-accfe598-5a7b-4ed0-a67c-66e11a36bd11.png)
每次渲染都有大量的内存生成

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/234038988-82813c03-9d05-4081-9073-9d65df76ecb0.png)
大量的对象和数组没有给释放

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/234039549-51b9b715-25f9-469f-8325-255ba8c13079.png)
生成的数据说这样的, 看着是node的每个节点对应的对象

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/234040080-9b9151dc-f778-4c08-884a-4af1c1cd2931.png)
懂得老铁帮忙研究下

**Reply by code-lixm** [reply]
canvas 没释放v4有这个bug
